### ISSUE TITLE
Copying textual content from Mattermost is poor executed

### ISSUE THREAD

**Issue by ylluminate** [mattermost/mattermost#11177]
#### Summary
When I copy content from a Mattermost channel and paste into another program (or within another Mattermost channel) I get very poorly formatted output.  Mattermost needs to have a differently formatted output for the copy / paste buffer when copying content that facilitates easier reading without the user having to reflow/reformat everything. 

#### Steps to reproduce
Copy content from any chat/channel within the web client.

#### Expected behavior
Pasted content should not be so poorly formatted.

#### Observed behavior (that appears unintentional) 
So to help with the context here, here's the results of a copy and paste of some content:
```
username1 profile image	
username1
2:38 PM



I'll send over the server login

what I can see is that there's a redirect

that goes from somedomain.tld to this anotherdomain.tld

username2 profile image	
username2
2:39 PM
but you're seeing the actual problem with anotherdomain.tld in terms of googlebot vs straight browsing, right?

ie, you're not only seeing the googlebot problem when you use earthflightaviation.com and it redirects to anotherdomain.tld

(even though you might still see a problem, it's not isolated to that scenario)

username1 profile image	
username1
2:40 PM
yes

I mean it CRAWLS the site

it's crawlable

and indexable

the bot is allowed

username2 profile image	
username2
2:40 PM
k, yeah, will need to see some more detailed info. do you have a login for the google webmaster for the anotherdomain.tld domain?

username1 profile image	
username1
2:40 PM
but it seems like it's acting like it's disallowing other bots

I should

waiting to get it from the team

username2 profile image	
username2
2:41 PM
k
```

As you can observe, there's extraneous or non-useful formatting and data in this.  Each "chunk" consists of:
```
username profile image
username
TIMESTAMP
````
Also, each line has an extra newline within the chat itself.

#### Possible fixes
It would be more useful if we were to format output like this (or allow us to define it):

```
username1 (TIMESTAMP): this is a line of text
username1 (TIMESTAMP): another line of text
username2 (TIMESTAMP): and yet another line of text
username3 (TIMESTAMP): more text
```

I personally like to have the timestamp + short date format within parentheses each time to maintain context, but the big thing is that everything could be squeezed together and made very clear and compact.  This is what, I think, anyone would really want out of a copy and paste of text content.

When we bring pictures into the mix, it probably should copy a URL to the image on the Mattermost server - or, if it's a link, don't include any rendered output that shows up in Mattermost and simply give the link(s).  Further, if there is an image, we might be alright to include a copy of the image, but we obviously don't want this on plaintext pastes.  

Personally I think we should first deal with making the text copy/paste function nice and then regroup on what we could or should do on the non-text data (images, formatting, etc.).

One other possibility is to make this copied content markdown friendly so as to retain the limited formatting that could be present such as bold, italics, etc.

**Comment by amyblais** [comment]
Hi @ylluminate, thank you for sharing this feedback. Would you like to share this on our feature request forum here (given this issue has probably existed since the beginning): https://mattermost.uservoice.com/forums/306457-general? If you're interested helping implement this, please let us know.

**Comment by ylluminate** [comment]
Thanks @amyblais. I guess I don't really feel like that forum get's enough exposure in some aspects or really prioritizes some things properly.  I would be interested in poking about on this issue as it's pretty important and I'm basically about to implement a filter myself in Pastebot or Keyboard Maestro to reformat these... It really shouldn't be necessary to do that and I'm guessing a fix would be relatively simple... at least I'd hope.

**Comment by amyblais** [comment]
I found there is a ticket open here for this: https://mattermost.atlassian.net/browse/MM-15727.

I'll add a comment asking whether it can be opened as Help Wanted.

**Comment by andrey-yantsen** [comment]
Hey @amyblais! Any news on this one? The current behaviour is extreeeeemely annoying...

**Comment by devicenull** [comment]
+1 This is absurdly bad right now... I can't really tell where this is being tracked, is it in github? jira? forums?

**Comment by ylluminate** [comment]
Yeah, it's such a sad situation for such an insanely useful system that is Mattermost. I truly wish someone could make this a priority.

**Comment by amyblais** [comment]
There is a PR open here: https://github.com/mattermost/mattermost-webapp/pull/3951.

**Comment by ylluminate** [comment]
Thanks again for calling that out @amyblais. It would be really great to see a minor dot release ASAP to get this active in the system. And also the additional remarks about more compressed formatting would be super helpful since it seems as though it should be easy to modify as per his remarks.

**Comment by amyblais** [comment]
I'm not sure if we're able to do a dot release for this as this is a feature request and would require soaking time and being part of full release testing. We hope that we would be able to finish the PR soon. Maybe we could consider looking into doing a custom build from the PR once it's completed.

**Comment by lieut-data** [comment]
Thanks for reporting this issue and for the community discussion around it. After multiple years of inactivity, we're closing this to help keep our issue tracker focused on active work. If this is still relevant, please feel free to open a new issue with updated context and we'll be happy to take another look.
